tests/test_text.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
```

Comment:
Can you confirm that all other tests are passinng? To make sure that our initial update for Text_Based is not breaking anything else. If you right-click on Tests folder, you can run unittest, and it runs everything

Reply:
Yay! :D

Reply:
I'm not able to run the tests. Some are passing, but mostly failing. Guessing this is due to recent changes.

Reply:
You should be able to run the tests in the new commits. Please let me know if you have any issue.